distribution/push_v2.go

```diff
@@ -240,10 +240,10 @@ func (pd *v2PushDescriptor) DiffID() layer.DiffID {
 }
 
 func (pd *v2PushDescriptor) Upload(ctx context.Context, progressOutput progress.Output) (distribution.Descriptor, error) {
-	if fs, ok := pd.layer.(layer.ForeignSourcer); ok {
-		if d := fs.ForeignSource(); d != nil {
+	if fs, ok := pd.layer.(distribution.Describable); ok {
+		if d := fs.Descriptor(); len(d.URLs) > 0 {
```

Comment:
Right, I think https://github.com/docker/docker/pull/23327/files#diff-77a3b9683ea5b78ddbf146c0cbad74e1R48 should check for just the mediaType instead of the URLs length also